content/blogposts/2022/openssl-307-vulnerabilities.md

```diff
@@ -11,81 +11,25 @@ heroImage:
-
-Log4j also botched their initial fixes quite badly, which resulted in more pain for Log4j users. It's not impossible that the fix in OpenSSL 3.0.7 will require further work, but it's conceptually simpler than the Log4j fix was, and my initial judgement is that it's likely to be correct and not require further work.
-
-Finally, upgrading OpenSSL 3 is likely to be considerably easier in most cases. Most users will use libraries that dynamically link to OpenSSL, which means that upgrading the OpenSSL library installed on the operating system will generally fix the issue. No changes to API calls or the like are required.
-
-
-## What can we do to help our customers mitigate this issue?
-
+## How can you mitigate the issue?
 
 ### Upgrades
 
-The general advice for OpenSSL 3 users will be to upgrade to OpenSSL 3.0.7 as soon as possible.
-
-For users who are only using OpenSSL transitively as a dynamic dependency (which will be most of them), this really means rebuilding container images and ensuring bare-metal servers are upgraded to the current version of OpenSSL. In general, most users will have processes in place for this already, and specific tools such as Renovate and Snyk will be able to provide insight into what needs to be done there that we shouldn't expend significant effort on trying to replicate.
+The general advice for OpenSSL 3 users is to upgrade to OpenSSL 3.0.7 as soon as possible.
 
+For users who are only using OpenSSL transitively as a dynamic dependency, this really means rebuilding container images and ensuring bare-metal servers are upgraded to the current version of OpenSSL. 
 
 ### Other mitigations
 
 OpenSSL recommends disabling TLS client authentication if it's not strictly necessary until an upgrade can be performed. [Per the FAQ](https://www.openssl.org/blog/blog/2022/11/01/email-address-overflows/):
 
-
-_Q: Are there any mitigations until I can upgrade?_
-
-
-_A: Users operating TLS servers may consider disabling TLS client authentication, if it is being used, until fixes are applied._
-
-This is, of course, unhelpful for users who actually need client authentication, but it's possible some users have accidentally enabled client authentication when they don't actually require it. Some initial thoughts on possible code searches are below.
-
-This may also affect users who have code that parses arbitrary server certificates. Detecting that is less a question of code search and is more a question of product knowledge.
-
-
 #### The low level API, and C and C++ users
 
 Client authentication is ultimately enabled by setting [the SSL_VERIFY_PEER option](https://www.openssl.org/docs/manmaster/man3/SSL_CTX_set_verify.html) when issuing or handling a TLS request. The problem is that this option is also used by clients for innocuous purposes, and there are no API differences between server and client mode — or, for that matter, vulnerable and non-vulnerable versions of OpenSSL — to craft a one-size-fits-all code search.
```

Comment:
@LawnGnome would you reframe this section, or pull it? Mainly “The problem is that this option is also used by clients for innocuous purposes, and there are no API differences between server and client mode” is giving me pause.

Reply:
The idea there is basically just messaging that "code search is somewhere between an overly blunt tool and completely useless in this context".

A simpler version of this would be something like:

> TLS client authentication is triggered by setting the `SSL_VERIFY_PEER` flag when handling a TLS request. However, this flag is also used by unrelated client requests, so searching for `SSL_VERIFY_PEER` is unhelpful on its own.